client/web/src/components/widgets/rulers/CanvasRuler.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+	<div class="canvas-ruler" :class="direction.toLowerCase()" ref="rulerRef">
+		<svg :style="svgBounds">
+			<path :d="svgPath" />
+		</svg>
+	</div>
+</template>
+
+<style lang="scss">
+.canvas-ruler {
+	flex: 1 1 100%;
+	background: var(--color-5-dullgray);
+	overflow: hidden;
+	position: relative;
+
+	&.vertical {
+		width: 16px;
+	}
+
+	&.horizontal {
+		height: 16px;
+	}
+
+	svg {
+		position: absolute;
+
+		path {
+			stroke-width: 1px;
+			stroke: var(--color-7-middlegray);
+		}
+	}
+}
+</style>
+
+<script lang="ts">
+import { defineComponent, PropType } from "vue";
+
+const RULER_THICKNESS = 16;
+
+export enum RulerDirection {
+	"Horizontal" = "Horizontal",
+	"Vertical" = "Vertical",
+}
+
+export default defineComponent({
+	props: {
+		direction: { type: String as PropType<RulerDirection>, default: RulerDirection.Vertical },
+		origin: { type: Number, required: true },
+		majorMarkSpacing: { type: Number, required: true },
+		mediumDivisions: { type: Number, default: 5 },
+		minorDivisions: { type: Number, default: 2 },
+	},
+	computed: {
+		svgPath(): string {
+			const isVertical = this.direction === RulerDirection.Vertical;
+			const lineDirection = isVertical ? "H" : "V";
+
+			let offsetStart = this.origin % this.majorMarkSpacing;
+			if (offsetStart < this.majorMarkSpacing) offsetStart -= this.majorMarkSpacing;
+
+			const divisions = this.majorMarkSpacing / this.mediumDivisions / this.minorDivisions;
+			const majorMarksFrequency = this.mediumDivisions * this.minorDivisions;
+
+			let d = "";
```

Comment:
I think `d` could have a more descriptive name

Reply:
I think 'd' here refers to the d attribute of svg elements. The d attribute defines a path to be drawn.
Not saying it couldn't be more descriptive, but I feel it is not too inapropriate.

Reply:
Ah, that make sense. Maybe a comment would be nice to clarify, but the name is probably acceptable. Alternatively, the variable could be named `path` or something, and a comment could indicate that it corresponds to the `d` attribute.